kotlin/openID4VP/src/commonMain/kotlin/io/mosip/openID4VP/common/Utils.kt

```
@@ -300,17 +307,44 @@ fun encodeToMultibaseBase58btc(bytes: ByteArray): String {
+                        ignoreCase = true
+                    ) || rawKey.algorithm.equals("EdDSA", ignoreCase = true)
+                ) {
+                    JWSAlgorithm.EdDSA
+                } else {
+                    throw IllegalArgumentException("Unsupported key type: ${rawKey.javaClass.simpleName}")
+                }
+            }
         }
+
+        jwsAlgorithm.name
+    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)